### ISSUE TITLE
Error to install build dependencies

### ISSUE THREAD

**Issue by CleyssonDias** [gunthercox/ChatterBot#2301]
My error is: 

Collecting chatterbot
  Using cached ChatterBot-1.0.5-py2.py3-none-any.whl (67 kB)
Collecting pymongo<4.0,>=3.3
  Using cached pymongo-3.13.0-cp310-cp310-win_amd64.whl (394 kB)
Collecting nltk<4.0,>=3.2
  Using cached nltk-3.8.1-py3-none-any.whl (1.5 MB)
Collecting mathparse<0.2,>=0.1
  Using cached mathparse-0.1.2-py3-none-any.whl (7.2 kB)
Collecting pytz
  Downloading pytz-2022.7.1-py2.py3-none-any.whl (499 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 499.4/499.4 kB 1.2 MB/s eta 0:00:00
Collecting pint>=0.8.1
  Using cached Pint-0.20.1-py3-none-any.whl (269 kB)
Collecting pyyaml<5.2,>=5.1
  Using cached PyYAML-5.1.2.tar.gz (265 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting python-dateutil<2.8,>=2.7
  Using cached python_dateutil-2.7.5-py2.py3-none-any.whl (225 kB)
Collecting sqlalchemy<1.3,>=1.2
  Using cached SQLAlchemy-1.2.19.tar.gz (5.7 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting spacy<2.2,>=2.1
  Using cached spacy-2.1.9.tar.gz (30.7 MB)
  Installing build dependencies ... error
  error: subprocess-exited-with-error

  × pip subprocess to install build dependencies did not run successfully.
  │ exit code: 1
  ╰─> [81 lines of output]
      Collecting setuptools
        Using cached setuptools-67.6.0-py3-none-any.whl (1.1 MB)
      Collecting wheel<0.33.0,>0.32.0
        Using cached wheel-0.32.3-py2.py3-none-any.whl (21 kB)
      Collecting Cython
        Using cached Cython-0.29.33-py2.py3-none-any.whl (987 kB)
      Collecting cymem<2.1.0,>=2.0.2
        Using cached cymem-2.0.7-cp310-cp310-win_amd64.whl (29 kB)
      Collecting preshed<2.1.0,>=2.0.1
        Using cached preshed-2.0.1.tar.gz (113 kB)
        Preparing metadata (setup.py): started
        Preparing metadata (setup.py): finished with status 'done'
      Collecting murmurhash<1.1.0,>=0.28.0
        Using cached murmurhash-1.0.9-cp310-cp310-win_amd64.whl (18 kB)
      Collecting thinc<7.1.0,>=7.0.8
        Using cached thinc-7.0.8.tar.gz (1.9 MB)
        Preparing metadata (setup.py): started
        Preparing metadata (setup.py): finished with status 'done'
      Collecting blis<0.3.0,>=0.2.1
        Using cached blis-0.2.4.tar.gz (1.5 MB)
        Preparing metadata (setup.py): started
        Preparing metadata (setup.py): finished with status 'done'
      Collecting wasabi<1.1.0,>=0.0.9
        Using cached wasabi-0.10.1-py3-none-any.whl (26 kB)
      Collecting srsly<1.1.0,>=0.0.6
        Using cached srsly-1.0.6-cp310-cp310-win_amd64.whl (196 kB)
      Collecting numpy>=1.7.0
        Using cached numpy-1.24.2-cp310-cp310-win_amd64.whl (14.8 MB)
      Collecting plac<1.0.0,>=0.9.6
        Using cached plac-0.9.6-py2.py3-none-any.whl (20 kB)
      Collecting tqdm<5.0.0,>=4.10.0
        Using cached tqdm-4.65.0-py3-none-any.whl (77 kB)
      Collecting colorama
        Using cached colorama-0.4.6-py2.py3-none-any.whl (25 kB)
      Installing collected packages: wasabi, plac, cymem, wheel, srsly, setuptools, preshed, numpy, murmurhash, Cython, colorama, tqdm, blis, thinc
        DEPRECATION: preshed is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
        Running setup.py install for preshed: started
        Running setup.py install for preshed: finished with status 'error'
        error: subprocess-exited-with-error

        Running setup.py install for preshed did not run successfully.
        exit code: 1

        [26 lines of output]
        C:\Users\cleys\AppData\Local\Programs\Python\Python310\lib\site-packages\setuptools\installer.py:27: SetuptoolsDeprecationWarning: setuptools.installer is deprecated. Requirements should be satisfied by a PEP 517 installer.
          warnings.warn(
        WARNING: The wheel package is not available.
        running install
        C:\Users\cleys\AppData\Local\Programs\Python\Python310\lib\site-packages\setuptools\command\install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
          warnings.warn(
        running build
        running build_py
        creating build
        creating build\lib.win-amd64-cpython-310
        creating build\lib.win-amd64-cpython-310\preshed
        copying preshed\about.py -> build\lib.win-amd64-cpython-310\preshed
        copying preshed\__init__.py -> build\lib.win-amd64-cpython-310\preshed
        creating build\lib.win-amd64-cpython-310\preshed\tests
        copying preshed\tests\test_counter.py -> build\lib.win-amd64-cpython-310\preshed\tests
        copying preshed\tests\test_hashing.py -> build\lib.win-amd64-cpython-310\preshed\tests
        copying preshed\tests\test_pop.py -> build\lib.win-amd64-cpython-310\preshed\tests
        copying preshed\tests\__init__.py -> build\lib.win-amd64-cpython-310\preshed\tests
        copying preshed\counter.pyx -> build\lib.win-amd64-cpython-310\preshed
        copying preshed\maps.pyx -> build\lib.win-amd64-cpython-310\preshed
        copying preshed\counter.pxd -> build\lib.win-amd64-cpython-310\preshed
        copying preshed\maps.pxd -> build\lib.win-amd64-cpython-310\preshed
        copying preshed\__init__.pxd -> build\lib.win-amd64-cpython-310\preshed
        running build_ext
        building 'preshed.maps' extension
        error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
        [end of output]

        note: This error originates from a subprocess, and is likely not a problem with pip.
      error: legacy-install-failure

      Encountered error while trying to install package.

      preshed

      note: This is an issue with the package mentioned above, not pip.
      hint: See above for output from the failure.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× pip subprocess to install build dependencies did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

help me

**Comment by woandms** [comment]
Install This it might fix it,
Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
    [end of output]

**Comment by bizbazboz** [comment]
You need to change your python interpreter to python 3.7.5 and install an older version of the ChatterBot Library with pip.
I did it with this command:
``` bash
pip install chatterbot==3.0.0
```

**Comment by magicoli** [comment]
> pip install chatterbot==3.0.0

This produces an error too, it looks like the latest version of chatterbot is way lower than 3.0

As said in https://github.com/gunthercox/ChatterBot/issues/2303#issuecomment-1502374512

I got the same issue. I solved it by installing python3.7 and setting up a virtual environment (as my default version is 3.10, and I need to keep it for other projects).

```
sudo apt install python3.7 python3.7-venv
python3.7 -m venv /path/to/chatterbot/folder/
cd /path/to/chatterbot/folder/
source ./bin/activate
pip install -U pip setuptools wheel
pip install chatterbot

```
While installation was successful, I still couldn't get any of the examples working, it seems not only the 3.7 requirement is missing from the README but also some other steps I can't figure out.
